frontend/src/metabase/lib/settings.js

```diff
@@ -88,9 +88,12 @@ class Settings {
   // Right now, all Metabase Cloud hosted instances run on *.metabaseapp.com
   // We plan on changing this to look at an envvar in the future instead.
   isHosted() {
-    return this.get("site-url")
-      .toLowerCase()
-      .includes("metabaseapp.com");
+    return (
+      this.get("site-url") &&
+      this.get("site-url")
+        .toLowerCase()
+        .includes("metabaseapp.com")
+    );
```

Comment:
```suggestion
    return /.+@metabase.com$/i.test(this.get("site-url"));
```

- This takes care of the case.
- If no url or if it's undefined, this will return `false` (no need for an additional check)
- More importantly, it makes sure that the we allow **only** `metabase.com` domain (addresses warning [Incomplete URL substring sanitization](https://github.com/metabase/metabase/pull/14511/checks?check_run_id=1765933701))

Reply:
If you find one-line solution too terse:

```javascript
isHosted() {
  const regex = new RegExp(/.+@metabase.com$/i)
  return regex.test(this.get("site-url"));
}
```

Reply:
The domain used for hosting is `<custom>.metabaseapp.com` - not `metabase.com` and it's not an email.

Reply:
Oops, yes. I was too tired last night :D
Thank you